base/Base.jl

```diff
@@ -35,6 +35,7 @@ getproperty(x::Tuple, f::Int) = (@inline; getfield(x, f))
 setproperty!(x::Tuple, f::Int, v) = setfield!(x, f, v) # to get a decent error
 
 getproperty(x, f::Symbol) = (@inline; getfield(x, f))
+getproperty(f::Union{Symbol,String,Int}) = Base.Fix2(getproperty, f)
```

Comment:
I think this could be `Any`; really only Symbol and Int are used but we don't restrict what you can accept as the second argument.

Reply:
She this be changed for `hasproperty` too?

Reply:
Is this a general concern about this PR or about changing this argument type to `Any`?

Reply:
It's general:
```
julia> xs = [(a=rand(), b=rand()) for _ in 1:100];

julia> @btime sum(x -> getproperty(x, :a), $xs);
  39.564 ns (0 allocations: 0 bytes)

julia> let g = Base.Fix2(getproperty, :a)
         @btime sum($g, $xs);
       end;
  3.792 μs (200 allocations: 4.69 KiB)
```

Reply:
This kind of gets into how much code specialization you want to do. If you know ahead of time that you just want to access `:a`, then one new anonymous function is the better choice. If the property name being accessed is determined programmatically or for looking up columns in a dynamically sized table, then you probably don't want to generate a bunch of new methods.